Unexpected panic occurs when time does not have a `-p` flag

Adventures into NixOS prove that the builtin `time` might not treat `-p` as a POSIX standard 😳

```
$ ./etime sleep 2
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/zimeg/emporia-time/pkg/times.parseTimeResults({0x0?, 0x2?})
        /home/ez/programming/etime/pkg/times/time.go:74 +0x29c
github.com/zimeg/emporia-time/pkg/times.TimeExec({{0xc000124130, 0x2, 0x2}, {{0x0, 0x0}, 0x0, {0x0, 0x0}, 0x0, {0x0, ...}, ...}})
        /home/ez/programming/etime/pkg/times/time.go:59 +0x29e
github.com/zimeg/emporia-time/cmd.Run({{0xc000124130, 0x2, 0x2}, {{0x0, 0x0}, 0x0, {0x0, 0x0}, 0x0, {0x0, ...}, ...}}, ...)
        /home/ez/programming/etime/cmd/etime.go:35 +0xbc
main.main()
        /home/ez/programming/etime/main.go:33 +0x2d8
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,6 +7,10 @@ to [Semantic Versioning][semver].
 
 ## [Unreleased]
 
+### Fixed
+
+- Avoid assumptions about the $PATH to the `time` command
+
 ### Maintenance
 
 - Enforce common linter checks before building the binary

--- a/pkg/times/stopwatch.go
+++ b/pkg/times/stopwatch.go
@@ -7,6 +7,7 @@ import (
 	"math/rand"
 	"os"
 	"os/exec"
+	"path/filepath"
 	"strings"
 	"time"
 )
@@ -34,6 +35,15 @@ func (bw *bufferWriter) Write(p []byte) (int, error) {
 
 // timerCommand forms the command struct for a timer to be parsed
 func timerCommand(command []string, stderr bufferWriter) *exec.Cmd {
+	timer, err := exec.LookPath("time")
+	if err != nil {
+		timer = "time"
+	} else {
+		timer, err = filepath.Abs(timer)
+		if err != nil {
+			timer = "time"
+		}
+	}
 	timeShell := []string{
 		strings.Join(command, " "),
 		";",
@@ -47,7 +57,7 @@ func timerCommand(command []string, stderr bufferWriter) *exec.Cmd {
 		"-c",
 		strings.Join(timeShell, " "),
 	}
-	cmd := exec.Command("/usr/bin/time", timeArgs...)
+	cmd := exec.Command(timer, timeArgs...)
 	if errors.Is(cmd.Err, exec.ErrDot) {
 		cmd.Err = nil
 	}

--- a/pkg/times/time.go
+++ b/pkg/times/time.go
@@ -71,6 +71,9 @@ func parseTimeResults(output string) (times CommandTime, err error) {
 	lines := strings.TrimSpace(output)
 	for _, line := range strings.Split(lines, "\n") {
 		fields := strings.Fields(line)
+		if len(fields) < 2 {
+			continue
+		}
 		measurement, value := fields[0], fields[1]
 		switch measurement {
 		case "real":

